repos/system_upgrade/common/actors/scansubscriptionmanagerinfo/tests/test_scansubscriptionmanagementinfo.py

```diff
@@ -0,0 +1,349 @@
+    assert command_runner.calls_performed_counter > 0, 'Some calls to subscription-manager were expected.'
+    assert actor_products.model_instances, 'The actor did not produce a message.'
+
+    produced_message = actor_products.model_instances[0]
+    assert produced_message.attached_skus, 'Actor failed to parse SKU.'
+    assert len(produced_message.attached_skus) == 1, \
+        'There was only one SKU in the mocked RHSM output, however more were detected by the actor.'
+    assert produced_message.attached_skus[0] == '598339696910', \
+        'The parsed SKU is different than the one contained in the mocked RHSM output.'
+
+
+def test_scansubscriptionmanagementinfo_get_available_repo_ids(monkeypatch, common_monkeypatched_handles):
+    """Tests whether the actor is able to get and parse avaiable repo ids."""
+
+    def mocked_get_parsed_repofiles(context):
+        repo_data_0 = RepositoryData(
+            repoid='rhel-8-for-x86_64-baseos-rpms',
+            name='Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+            enabled=True)
+
+        repo_data_1 = RepositoryData(
+            repoid='satellite-tools-6.7-for-rhel-8-x86_64-rpms',
+            name='Satellite',
+            enabled=True)
+
+        repo_data_2 = RepositoryData(
+            repoid='rhel-7-for-x86_64-baseos-rpms',
+            name='Red Hat Enterprise Linux 7 for x86_64 - BaseOS (RPMs)',
+            enabled=True)
+
+        return [
+            RepositoryFile(file='/test', data=[repo_data_2]),
+            RepositoryFile(file='/etc/yum.repos.d/redhat.repo', data=[repo_data_0, repo_data_1])]
+
+    actor_products = common_monkeypatched_handles[1]
+
+    monkeypatch.setattr(repofileutils, 'get_parsed_repofiles', mocked_get_parsed_repofiles)
+
+    scanrhsm.scan()
+
+    assert actor_products.model_instances, \
+        'The actor did not produce a message.'
+    produced_message = actor_products.model_instances[0]
+    assert len(produced_message.available_repos) == 2, \
+        'The actor produced a message with different count of available repositories than expected.'
+
+    available_repo_ids = produced_message.available_repos
+    assert 'rhel-8-for-x86_64-baseos-rpms' in available_repo_ids, \
+        'The actor did not report expected repository id.'
+    assert 'satellite-tools-6.7-for-rhel-8-x86_64-rpms' in available_repo_ids, \
+        'The actor did not report expected repository id.'
+
+
+def test_scansubscriptionmanagementinfo_with_skip_rhsm(monkeypatch, common_monkeypatched_handles):
+    """Tests whether the actor respects the LEAPP_NO_RHSM environmental variable."""
+
+    _, actor_products, external_command_runner = common_monkeypatched_handles
+
+    _original_get_env = rhsm.get_env
+
+    def mocked_get_env(env_variable, default):
+        if env_variable == 'LEAPP_NO_RHSM':
+            return '1'
+        return _original_get_env(env_variable, default)
+
+    monkeypatch.setattr(rhsm, 'get_env', mocked_get_env)
```

Comment:
replace by  `MockedCurrentActor` (see the `envars` parameter)

Reply:
Thanks for the tip :+1:  (new changes/commits already doesn't mock `get_env`).